api/server.spec.js

```diff
@@ -0,0 +1,89 @@
+const request = require("supertest");
+const server = require("../api/server");
+
+const users = require("../models/users.js");
+const db = require("../database/config/index.js");
+
+/* ---------------------- User Endpoints ---------------*/
+
+// beforeEach(async () => {
+//     // this function executes and clears out the table before each test
+//     await db('couples').truncate();
+// });
+
+describe('GET /', () => {
+    it('has process.env.DB_ENV as "testing"', () => {
+        expect(process.env.DB_ENV).toBe('testing');
+    })
+})
+
+
+describe('/api/users CRUD', function() {
+    it('should create a user', async function(done) {
+        request(server)
+            .post('/api/users')
+            .send({
+                    spouse_one_name: "Gaston",
+                    spouse_two_name: "Linda",
+                    email: "gastonandlinda@gmail.com",
+                    password: "donfftgetmarriedagainguyspleaseeee"
+                })
+            .set('Accept', 'application/json')
+            .expect(201)
+            done()
+           
+        });
+    
+
+    it('should return 500 if missing spouse names,email,password fields of user', function(done) {
+        request(server)
+            .post('/api/users')
+            .send({ spouse_one_name: '', spouse_two_name: null, email: '', password: '' })
+            .expect(500, done)
+            
+    });
+
+    it('it should update a user', async function(done) {
+        request(server)
+            .put('/api/users/1')
+            .send({
+                spouse_one_name: "Robert",
+                spouse_two_name: "Jane",
+                email: "robertandjane@gmail.com",
+                password: "partytime123"
+            })
+            .expect(200)
+            done()
+       
+    });
+
+    it('should return the user table with ID', async function(done) { 
+        request(server)
+            .get('/api/users/1')
+            .expect(200)
+            done()
+    });
+    
+    it('should delete a user', async function(done) {
+            request(server)
+                .del('/api/users/58')
```

Comment:
Does this test work everytime you run it? I'd be surprised since we don't have 58 seeded users.

If it does work, I think it's because there's an issue with the Users models/router. Will need to investigate further.